Sources/StreamChatUI/CommonViews/AvatarView/ChatPresenceAvatarView/ChatPresenceAvatarView.swift

```diff
@@ -35,4 +43,26 @@ open class _ChatPresenceAvatarView<ExtraData: ExtraDataTypes>: _View, UIConfigPr
             .pin(equalTo: widthAnchor, multiplier: 0.3)
             .isActive = true
     }
+    
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+        
+        setUpMask(indicatorVisible: showOnlineIndicator)
+    }
+    
+    /// Creates space for indicator view in avatar view by masking path provided by the indicator view.
+    /// - Parameter visible: Bool to determine if the indicator should be shown. The avatar view won't be masked if the indicator is not visible.
+    open func setUpMask(indicatorVisible: Bool) {
+        guard
+            indicatorVisible,
```

Comment:
Why not just check `onlineIndicatorView.isVisible`?

Reply:
I don't want to depend on `onlineIndicatorView.isVisible` because then the `setUpMask` function would always have to be called *after* you set the `onlineIndicatorView.isVisible`, giving the consumer a bigger chance to make a mistake by changing the call order.